### DISCUSSION TITLE
ltsp / guest / onedrive

### DISCUSSION THREAD

**Discussion by stiebel** [ltsp/ltsp#518] [Ideas]
Hello again,
I have been pushed away as the system admin for a year by an "expert" installing windows on all our machines.
Now the users keep moaning about low speed and inflexibility...

So I will get back a server to run a linux system for those who want to change, but

I do not have access to the AD of the windows system and I had to promise not to use it at all
to not "mix up everything".
Please do not discuss the sense of this, that's politics and I cannot change the situation.
I am also not allowed to make any changes at the dhcp.
The only change I am allowed to make is to enable USB-boot

So here is my plan:

I will use ipxe from a bootable USB-Stick 
With this ipxe I will chain-load an ltsp image from an ltsp server
This image shall load and start with an auto-login to a guest account 
This guest account shall be as open as possible, but with no save-to-disk at all.

In this guest account I want to mount the users onedrive (with https://github.com/jstaf/onedriver)
This is no sync but a mount and the users have to "login" at this point.

Now the users have a running linux system AND their files to work with.
This should work in theory.

But then I know my users, they want to have their firefox-setting, backgrounds, custom mouse speed and so on.
So is there a chance to mount the onedrive and after that remap the users home-dir to a onedrive subfolder and reload all the settings from there and automatically store the custom settings there as well? 

This may sound like a weird solution, but think about it.
This would give the chance for an "almoust zero efford" but flexible linux-alternative within a closed windows domain.
If a user screwed up all the settings he just has to delete this subfolder and can "start from scratch" without affecting his data.

Thanks for your help
Roland

**Comment by alkisg** [comment]
> So is there a chance to mount the onedrive and after that remap the users home-dir to a onedrive subfolder and reload all the settings from there and automatically store the custom settings there as well?

Sure all the things you write are possible. But note that you'll need some automation and bash scripting to mount onedrive, to save and restore the firefox settings etc, and LTSP can't help you with that. LTSP manages system settings via ltsp.conf, but not user settings.

If you don't have the skills to do that, a much easier alternative would be to not have guest accounts, but real Linux accounts (with SSSD/AD authentication), with real Linux settings and "Linux-side" documents, even if you do offer them access to onedrive for their "Windows-side" documents.

It will also be a whole lot faster and more stable; imagine what chaos could happen if your backup/restore script did something wrong and needed to transfer GBs of data over a slow or failed Internet connection...

**Reply by stiebel** [reply]
Hi alkisg,
Thanks for your fast answer.
You are absolutely right with all your points.
The reason I do not want to use real Linux accounts is, that I have 1000ish users with a change of 10% each year and I absolutely do not want to keep up this double data management. I am strictly not allowed to synchronize data with the AD.
But thinking about it, I also do not like the "settings-on-ondrive" idea.
If I use an USB-pen to boot with ipxe, I could simply use the USB drive as a place for the home directory.
Is there a ready-made solution for this? 
Then I would not have to reinvent the wheel for that.

**Comment by alkisg** [comment]
Start with "how do you manage the 1000ish user accounts"?

What I proposed was that they would authenticate against AD, so that you wouldn't have to manage them; they'd be created/updated/deleted in AD.
I'm talking about user accounts (username/password), NOT about user data (documents), that you said you can't synchronize.

If you don't do that, then you'd either need static accounts (a1-01, a1-02, b1-12...) or a signup process.

Sure, you can have /home mounted from a USB drive. But first test what device you want to buy, as some USB sticks are slow, and others quickly wear out after many writes (and /home does need frequent writes). External SSD disks would be a lot better but more expensive.

**Reply by stiebel** [reply]
Many thanks for your advises.
I will test several options and will post here, what I did.
c.u.
Roland
